core/src/commonMain/kotlin/com/ustadmobile/core/controller/ContentEntryDetailPresenter.kt

```diff
@@ -0,0 +1,69 @@
+package com.ustadmobile.core.controller
+
+import com.ustadmobile.core.db.UmAppDatabase
+import com.ustadmobile.core.util.safeParse
+import com.ustadmobile.core.view.*
+import com.ustadmobile.door.DoorLifecycleOwner
+import com.ustadmobile.door.doorMainDispatcher
+import com.ustadmobile.lib.db.entities.ContentEntry
+import com.ustadmobile.lib.db.entities.UmAccount
+import kotlinx.coroutines.*
+import org.kodein.di.DI
+
+class ContentEntryDetailPresenter(context: Any,
+                                  arguments: Map<String, String>, view: ContentEntryDetailView, di: DI,
+                                  lifecycleOwner: DoorLifecycleOwner)
+    : UstadDetailPresenter<ContentEntryDetailView, ContentEntry>(context, arguments, view, di, lifecycleOwner) {
+
+
+    override suspend fun onCheckEditPermission(account: UmAccount?): Boolean {
+        return false
+    }
+
+    override val persistenceMode: PersistenceMode
+        get() = PersistenceMode.DB
+
+    override fun onLoadFromJson(bundle: Map<String, String>): ContentEntry? {
+        super.onLoadFromJson(bundle)
+
+        val entityJsonStr = bundle[UstadEditView.ARG_ENTITY_JSON]
+        var editEntity: ContentEntry? = null
+        if(entityJsonStr != null) {
+            editEntity = safeParse(di,  ContentEntry.serializer(), entityJsonStr)
+        }else {
+            editEntity = ContentEntry()
+        }
+
+        GlobalScope.launch {
```

Comment:
There is no need for setupTabs to be suspended. This was done with other presenters where doing a database query was required to see what tabs were going to be displayed. The tab layout here is always the same. The GlobalScope.launch can be removed.